### ISSUE TITLE
Could we add a callback for slideToIndex ?

### ISSUE THREAD

**Issue by wszgxa** [xiaolin/react-image-gallery#375]
Our situation: 

We want to slide to the first image, but the animation is not so cool. So we decide to add a mask. After slide event finish we close the mask...

But slideToIndex will call `setState`, we know setState is asynchronous event. So I can't know when the slide event finish...

Could we add a callback to slideToIndex?

I can help to create the PR, if you are ok with this.

**Comment by xiaolin** [comment]
There is a callback when the image slides, you can use `onSlide` which is called after index has changed.

**Comment by xiaolin** [comment]
@wszgxa I see, you need the `onSlide` to be called after the animation transition. I think if you move
```jsx
      if (this.props.onSlide) {
        this.props.onSlide(this.state.currentIndex);
      }
```
after https://github.com/xiaolin/react-image-gallery/blob/master/src/ImageGallery.jsx#L354
it should do the trick. Feel free to open a pr.

**Comment by wszgxa** [comment]
Make sense.

**Comment by wszgxa** [comment]
Hmm, @xiaolin. Could you please publish a version with the change?

**Comment by xiaolin** [comment]
Sure, I will publish a new version this week.

**Comment by xiaolin** [comment]
@wszgxa published the new version today @ 0.8.13
